src/lightning/fabric/strategies/xla.py

```diff
@@ -222,12 +221,9 @@ def _set_world_ranks(self) -> None:
         rank_zero_only.rank = self.cluster_environment.global_rank()
 
     @staticmethod
-    def _validate_dataloader(dataloaders: DataLoader) -> None:
-        def check_has_len(dataloader: DataLoader) -> None:
-            if not has_len(dataloader):
-                raise TypeError(
-                    "TPUs do not currently support IterableDataset objects, the dataset must implement `__len__`."
-                    " HINT: You can mock the length on your dataset to bypass this MisconfigurationException."
-                )
-
-        apply_to_collection(dataloaders, dtype=object, wrong_dtype=(Sequence, Mapping), function=check_has_len)
```

Comment:
This makes sense. In fabric, this was redundant. This was overlooked when adapting the PL code to the fabric strategy.